Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1505,21 +1509,58 @@ function createAnimatedComponent(Component: any): any {
 
     componentWillUnmount() {
       this._propsAnimated && this._propsAnimated.__detach();
+      this._detachNativeEvents(this.props);
     }
 
     setNativeProps(props) {
       this._component.setNativeProps(props);
     }
 
     componentWillMount() {
-      this.attachProps(this.props);
+      this._attachProps(this.props);
     }
 
     componentDidMount() {
       this._propsAnimated.setNativeView(this._component);
+
+      this._attachNativeEvents(this.props);
+    }
+
+    _attachNativeEvents(newProps) {
+      if (newProps !== this.props) {
```

Comment:
This can perhaps be optimised so that we don't reattach native props every time when non relevant property gets changed. Also, I'm not sure if this statement can even evaluate to true (if props hasn't been updated `componentWillReceiveProps` will not be triggered)